go/libraries/doltcore/sqle/dprocedures/dolt_checkout.go

```diff
@@ -35,57 +35,108 @@ var ErrEmptyBranchName = errors.New("error: cannot checkout empty string")
 
 // doltCheckout is the stored procedure version for the CLI command `dolt checkout`.
 func doltCheckout(ctx *sql.Context, args ...string) (sql.RowIter, error) {
-	res, err := doDoltCheckout(ctx, args)
+	res, message, err := doDoltCheckout(ctx, args)
 	if err != nil {
 		return nil, err
 	}
-	return rowToIter(int64(res)), nil
+	return rowToIter(int64(res), message), nil
 }
 
-func doDoltCheckout(ctx *sql.Context, args []string) (int, error) {
+func generateSuccessMessage(newBranch, upstream string) string {
+	result := fmt.Sprintf("Switched to branch '%s'", newBranch)
+	if upstream != "" {
+		result += fmt.Sprintf("\nbranch '%s' set up to track '%s'.", newBranch, upstream)
+	}
+	return result
+}
+
+// willModifyDb determines whether or not this operation is a no-op and can return early with a helpful message.
+func willModifyDb(dSess *dsess.DoltSession, data env.DbData, dbName, branchName string, updateHead bool) (bool, error) {
+	headRef, err := data.Rsr.CWBHeadRef()
+	// If we're in a detached head state, allow checking out a new branch.
+	if err == doltdb.ErrOperationNotSupportedInDetachedHead {
+		return true, nil
+	}
+	if err != nil {
+		return false, err
+	}
+
+	// If the operation won't modify either the active session or the default session, return early.
+	isModification := headRef.GetPath() != branchName
+	if updateHead {
+		fs, err := dSess.Provider().FileSystemForDatabase(dbName)
+		if err != nil {
+			return false, err
+		}
+		repoState, err := env.LoadRepoState(fs)
+		if err != nil {
+			return false, err
+		}
+		defaultBranch := repoState.CWBHeadRef().GetPath()
+		isModification = isModification || (defaultBranch != branchName)
+	}
+	return isModification, nil
+}
+
+func doDoltCheckout(ctx *sql.Context, args []string) (int, string, error) {
```

Comment:
gDoc. Function name makes it pretty clear what this does, but what are the outputs?

Reply:
Added return variable names.